packages/components/src/FocusTrap.test.ts

```diff
@@ -0,0 +1,30 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { button, h } from 'react-hyperscript-helpers';
+
+import { FocusTrap } from './FocusTrap';
+
+describe('FocusTrap', () => {
+  it('traps focus', async () => {
+    // Arrange
+    const user = userEvent.setup();
+
+    const { container } = render(h(FocusTrap, [button(['A']), button(['B']), button(['C'])]));
```

Comment:
really nice test, but how about we add a Button 'D' outside the focus-trap to really validate that the trap is functioning.

Reply:
Excellent idea. Done in https://github.com/DataBiosphere/terra-ui/pull/4282/commits/7aed1dfef0c5d47f8b08446f605230c9a244c2eb.